bridge-sdk/connectors/omni-connector/src/omni_connector.rs

```diff
@@ -1,828 +1,285 @@
-            .map_err(|_| BridgeSdkError::UnknownError)?;
-
-        let tx_id = near_rpc_client::change(
-            near_endpoint,
-            self.near_signer()?,
-            token_locker_id.to_string(),
-            "deploy_token".to_string(),
-            serialized_args,
-            120_000_000_000_000,
-            4_000_000_000_000_000_000_000_000,
-        )
-        .await?;
-
-        tracing::info!(
-            tx_hash = format!("{:?}", tx_id),
-            "Sent deploy token transaction"
-        );
-
-        Ok(tx_id)
-    }
-
-    /// Transfers NEP-141 tokens to the token locker. The proof from this transaction is then used to mint the corresponding tokens on Ethereum
-    #[tracing::instrument(skip_all, name = "NEAR INIT TRANSFER")]
-    pub async fn near_init_transfer(
-        &self,
+        receiver: String,
+    },
+    EvmInitTransfer {
+        chain_kind: ChainKind,
         near_token_id: String,
         amount: u128,
         receiver: String,
-    ) -> Result<CryptoHash> {
-        let near_endpoint = self.near_endpoint()?;
-        let token_locker = self.token_locker_id()?.to_string();
-
-        let required_balance = self
-            .get_required_balance_for_init_transfer(&receiver, self.near_account_id()?.as_str())
-            .await?
-            + self.get_required_balance_for_account().await?;
-        let existing_balance = self.get_storage_balance().await?;
-
-        if existing_balance < required_balance {
-            self.storage_deposit(required_balance - existing_balance)
-                .await?;
-        }
-
-        let fee = 0;
-        let args =
-            format!(r#"{{"receiver_id":"{token_locker}","amount":"{amount}","msg":"{{\"recipient\":\"{receiver}\",\"fee\":\"{fee}\",\"native_token_fee\":\"0\"}}"}}"#)
-                .into_bytes();
-
-        let tx_hash = near_rpc_client::change(
-            near_endpoint,
-            self.near_signer()?,
-            near_token_id,
-            "ft_transfer_call".to_string(),
-            args,
-            300_000_000_000_000,
-            1,
-        )
-        .await?;
+        fee: Fee,
+    },
+}
 
-        tracing::info!(
-            tx_hash = format!("{:?}", tx_hash),
-            "Sent transfer transaction"
-        );
+pub enum FinTransferArgs {
+    NearFinTransfer {
+        chain_kind: ChainKind,
+        storage_deposit_actions: Vec<StorageDepositAction>,
+        prover_args: Vec<u8>,
+    },
+    EvmFinTransfer {
+        chain_kind: ChainKind,
+        event: Nep141LockerEvent,
+    },
+    EvmFinTransferWithTxHash {
+        chain_kind: ChainKind,
+        near_tx_hash: CryptoHash,
+    },
+}
 
-        Ok(tx_hash)
+impl OmniConnector {
+    pub fn new() -> Self {
+        Self::default()
     }
 
-    /// Mints the corresponding bridged tokens on Ethereum. Requires an MPC signature
-    #[tracing::instrument(skip_all, name = "EVM FIN TRANSFER")]
-    pub async fn evm_fin_transfer(
+    pub async fn evm_deploy_token(
```

Comment:
Done, I abstracted log_metadata, deploy_token and init/fin_transfer

Reply:
Do we need to keep these methods public then? If user needs evm-specific methods they can use a client instead?

Reply:
@kiseln I don't see a reason in making any method private in a sdk. For example, person can prefer using a direct method instead of an abstraction and I don't want to limit anyone